csrc/multidevice/communicator.h

```diff
@@ -31,8 +31,8 @@ namespace nvfuser {
 
 using RankType = DeviceIdxType;
 
-// Supported backends. TODO: only tested with nccl for now
-enum class CommunicatorBackend { nccl, ucc, gloo };
+// Supported backends. TODO: gloo untested
+enum class CommunicatorBackend { nccl, ucc, gloo, none };
```

Comment:
What about renaming `none` with `default`?
Actually, maybe we don't need to add a new entry to CommunicatorBackend, but rather to make the backend type argument optional inside the methods that use it
What do you think ?

Reply:
I think optional arguments will be the cleanest. Looks like Fuser mostly uses std::optional instead of c10::optional so I'll use std.